### ISSUE TITLE
ceph cluster ipv6 only failed

### ISSUE THREAD

**Issue by fbarbeira** [ceph/ceph-ansible#1569]
I'm trying to deploy a ipv6 only cluster but the playbook seems to fail at this point:

```
TASK [ceph.ceph-common : generate ceph configuration file: ceph.conf] **********************************************************************************************************************************************************************
[DEPRECATION WARNING]: ansible.utils.unicode.to_bytes is deprecated.  Use ansible.module_utils._text.to_bytes instead.
This feature will be removed in version 2.4. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: ansible.utils.unicode.to_unicode is deprecated.  Use ansible.module_utils._text.to_text instead.
This feature will be removed in version 2.4. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: ansible.utils.unicode.to_bytes is deprecated.  Use ansible.module_utils._text.to_bytes instead.
This feature will be removed in version 2.4. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: ansible.utils.unicode.to_unicode is deprecated.  Use ansible.module_utils._text.to_text instead.
This feature will be removed in version 2.4. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: ansible.utils.unicode.to_bytes is deprecated.  Use ansible.module_utils._text.to_bytes instead.
This feature will be removed in version 2.4. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: ansible.utils.unicode.to_unicode is deprecated.  Use ansible.module_utils._text.to_text instead.
This feature will be removed in version 2.4. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
fatal: [ceph-mon01]: FAILED! => {"failed": true, "msg": "'list object' has no attribute 'address'"}
fatal: [ceph-mon02]: FAILED! => {"failed": true, "msg": "'list object' has no attribute 'address'"}
fatal: [ceph-mon03]: FAILED! => {"failed": true, "msg": "'list object' has no attribute 'address'"}
	to retry, use: --limit @/home/ceph-deploy/ceph-ansible/site.retry

PLAY RECAP *********************************************************************************************************************************************************************************************************************************
```

This is an ifconfig of 'ceph-mon01':

```
# ifconfig eth0
eth0      Link encap:Ethernet  HWaddr 00:16:3e:xx:xx:xx
          inet6 addr: fe80::xxx:xxxx:xxxx:x/64 Scope:Link
          inet6 addr: xxxx:xxx:0:xx:xxx:xx:xxx:xxx/64 Scope:Global
          UP BROADCAST RUNNING MULTICAST  MTU:9000  Metric:1
          RX packets:7459 errors:0 dropped:0 overruns:0 frame:0
          TX packets:4770 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:9291393 (9.2 MB)  TX bytes:1382439 (1.3 MB)

#

```

Maybe the script expects that the destination machine must have an ipv4?

**Comment by leseb** [comment]
Did you solve your problem?
Was this really a bug from ceph-ansible or an infrastructure bug?

**Comment by fbarbeira** [comment]
After read the docs I established the 'alls.yml' file with the following:

```
# cat group_vars/all.yml | egrep -v ^# | grep -v ^$
---
dummy:
ceph_origin: 'upstream'
ceph_stable: true
ip_version: ipv6
radosgw_civetweb_bind_ip: "[{{ ansible_default_ipv6.address }}]"
nfs_file_gw: false
nfs_obj_gw: false
#
```

But it failed:

```
TASK [ceph-mon : ceph monitor mkfs with keyring] *******************************************************************************************************************************************************************************************
fatal: [ceph-mon01]: FAILED! => {"changed": true, "cmd": ["ceph-mon", "--cluster", "ceph", "--setuser", "ceph", "--setgroup", "ceph", "--mkfs", "-i", "ceph-mon01", "--fsid", "9b38f19b-03c6-4e36-aa72-6611ccaa322f", "--keyring", "/var/lib/ceph/tmp/keyring.mon.ceph-mon01"], "delta": "0:00:00.025205", "end": "2017-05-30 10:24:41.019384", "failed": true, "rc": 1, "start": "2017-05-30 10:24:40.994179", "stderr": "2017-05-30 10:24:41.016389 7f3f26590840 -1 unable to find any IP address in networks: 0.0.0.0/0", "stderr_lines": ["2017-05-30 10:24:41.016389 7f3f26590840 -1 unable to find any IP address in networks: 0.0.0.0/0"], "stdout": "", "stdout_lines": []}
fatal: [ceph-mon02]: FAILED! => {"changed": true, "cmd": ["ceph-mon", "--cluster", "ceph", "--setuser", "ceph", "--setgroup", "ceph", "--mkfs", "-i", "ceph-mon02", "--fsid", "9b38f19b-03c6-4e36-aa72-6611ccaa322f", "--keyring", "/var/lib/ceph/tmp/keyring.mon.ceph-mon02"], "delta": "0:00:00.025445", "end": "2017-05-30 10:24:41.089045", "failed": true, "rc": 1, "start": "2017-05-30 10:24:41.063600", "stderr": "2017-05-30 10:24:41.086077 7f1193a59840 -1 unable to find any IP address in networks: 0.0.0.0/0", "stderr_lines": ["2017-05-30 10:24:41.086077 7f1193a59840 -1 unable to find any IP address in networks: 0.0.0.0/0"], "stdout": "", "stdout_lines": []}
fatal: [ceph-mon03]: FAILED! => {"changed": true, "cmd": ["ceph-mon", "--cluster", "ceph", "--setuser", "ceph", "--setgroup", "ceph", "--mkfs", "-i", "ceph-mon03", "--fsid", "9b38f19b-03c6-4e36-aa72-6611ccaa322f", "--keyring", "/var/lib/ceph/tmp/keyring.mon.ceph-mon03"], "delta": "0:00:00.024213", "end": "2017-05-30 10:24:41.124057", "failed": true, "rc": 1, "start": "2017-05-30 10:24:41.099844", "stderr": "2017-05-30 10:24:41.121191 7f89d990d840 -1 unable to find any IP address in networks: 0.0.0.0/0", "stderr_lines": ["2017-05-30 10:24:41.121191 7f89d990d840 -1 unable to find any IP address in networks: 0.0.0.0/0"], "stdout": "", "stdout_lines": []}
```

Because the ansible provision **ceph.conf** with the following:

```
root@ceph-mon01:/etc/ceph# cat ceph.conf | grep network
public network = 0.0.0.0/0
cluster network = 0.0.0.0/0
root@ceph-mon01:/etc/ceph#
```

**Comment by leseb** [comment]
@fbarbeira You need to set  `public_network` to the right subnet.
We actually moved this check in ceph-osd but we should move it to ceph-mon instead.

**Comment by fbarbeira** [comment]
The ``public_network`` variable belongs to ``[global]`` section:

http://docs.ceph.com/docs/master/rados/configuration/network-config-ref/#public-network

Should I specify ``public_network`` ? If my cluster has no separated networks, I assume that is not necesary to especify both networks.

**Comment by leseb** [comment]
Yes please specify it.

**Comment by guits** [comment]
It should be fixed by #1594
